dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -124,13 +131,49 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
         THROW_HR_IF(E_UNEXPECTED, notificationId == 0);
         toast.ToastId(notificationId);
     }
-    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData /* data */, hstring /* tag */, hstring /* group */)
+
+    // API updates notification data when both tag and group are provided or when tag is provided but group is empty.
+    // For other scenarios the API returns ToastProgressResult::Failed.
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData data, hstring tag, hstring group)
     {
-        throw hresult_not_implemented();
+        std::wstring appId{ RetrieveToastAppId() };
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::IToastProgressData> toastProgressData
+            = winrt::make_self<ToastProgressDataABI>(data);
+
+        auto hr = ToastNotifications_UpdateNotificationData(appId.c_str(), tag.c_str(), group.c_str(), toastProgressData.get());
```

Comment:
Perhaps `auto hr{ LOG_IF_FAILED(ToastNotification_UpdateNotificationdata(...)) };` ?

Below you swallow `hr` and map it to Succeeded | NotFound | Failed so you lose the actual error details if something was wrong. `LOG_IF_FAILED` would help in that case.

Reply:
Actually it might not be needed now. We will be throwing hr if Failed scenario. Thanks.